docs/source/locality.rst

```diff
@@ -0,0 +1,80 @@
+Data Locality
+=============
+
+*Data movement often needlessly limits performance.*
+
+This is especially true for analytic computations.  ``Distributed`` minimizes
+data movement when possible and enables the user to take control when
+necessary.  This document describes current scheduling policies and user API
+around data locality.
+
+Current Policies
+----------------
+
+Task Submission
+```````````````
+
+In the common case distributed runs tasks on workers that already hold
+dependent data.  If you have a task ``f(x)`` that requires some data ``x`` then
+that task will very likely be run on the worker that already holds ``x``.
+
+If a task requires data split among multiple workers, then the scheduler chooses
+to run the task on the worker that requires the least data transfer to it.
+The size of each data element is measured by the workers using the
+``sys.getsizeof`` function, which depends on the ``__sizeof__`` protocol
+generally available on most relevant Python objects.
+
+Data Scatter
+````````````
+
+When a user scatters data from their local process to the distributed network
+this data is distributed in a round-robin fashion grouping by number of cores.
+So for example If we have two workers ``Alice`` and ``Bob``, each with two
+cores and we scatter out the list ``range(10)`` as follows:
+
+.. code-block:: python
+
+   futures = e.scatter(range(10))
+
+Then Alice and Bob receive the following data
+
+*  Alice: ``[0, 1, 4, 5, 8, 9]``
+*  Bob: ``[2, 3, 6, 7]``
+
+
+User Control
+------------
+
+Complex algorithms may require more user control.
+
+For example the existence of specialized hardware such as GPUs or database
+connections may restrict the set of valid workers for a particular task.
+
+In these cases use the ``workers=`` keyword argument to the ``submit``,
+``map``, or ``scatter`` functions as follows:
+
+.. code-block:: python
+
+   future = e.submit(func, *args, workers=['hostname1', 'hostname2'])
+
+Required data will always be moved to these workers, even if the volume of that
```

Comment:
ideally show some output (like you did above for Alice/Bob), e.g. how the data actually gets distributed (you may need a slightly more complex example). further would be nice to keep the continuity (e.g. maybe make Alice and Bob hostnames and add a 3rd to show what happens)